api/all/src/main/java/io/opentelemetry/api/trace/propagation/W3CTraceContextPropagator.java

```diff
@@ -245,6 +245,11 @@ private static TraceState extractTraceState(String traceStateHeader) {
       checkArgument(index != -1, "Invalid TraceState list-member format.");
       traceStateBuilder.put(listMember.substring(0, index), listMember.substring(index + 1));
     }
-    return traceStateBuilder.build();
+    TraceState traceState = traceStateBuilder.build();
+    if (traceState.size() != listMembers.length) {
+      // Validation failure, drop the tracestate
```

Comment:
Foes this happen only with duplicate keys?

Reply:
This happens for any sort of validation error, if a key or value doesn't validate we don't populate it into TraceState

Reply:
Ah, because `build` silently discards it? Makes sense.